### PR TITLE
add features && fix some bugs

### PR DESCRIPTION
**Pull request by wonderflow**
## Changes
   
  - [x] fix read line deadlock on wait for file exist
  - [x] refactor runner reset
  - [x] make cloudtrail reader resetable
  - [x] fix cloudtrail read .metastore  too frequency
  - [x] support .csv.zip for cloudtrial reader 
  - [x] add trim transformer
  - [x] support skip first line in dir mode file reader
  - [x] fix disable_web
  - [x] add magic number lag duration support for sql reader
  - [x] use use runnerStatus clone instead of deepcopyByJson
 
## Reviewers

  - [ ] @andrewei1316  please review
  - [ ] @redHJ  please review
  - [ ] @xingfeng2510 

## Wiki Changes
 
  - options1...
  - options2...
    
## Checklist
   
   - [ ] Rebased/mergable
   - [ ] Tests pass
   - [ ] Wiki updated

### CODE REVIEW

File: mgr/mgr.go [hunk 1]
```diff
@@ -773,27 +776,30 @@ func (m *Manager) ResetRunner(name string) (err error) {
...
+	// 出错的话，回滚并报错
+	resetErr := runnerReset.Reset()
+	if resetErr != nil {
+		log.Errorf("reset runner %v error %v", filename, resetErr)
+		// 此处就算失败也了不能直接return，需要回滚
 	}
 	conf.IsStopped = false
-	if err = m.ForkRunner(filename, conf, true); err != nil {
-		return fmt.Errorf("forkRunner %v error %v", filename, err)
+	err = m.ForkRunner(filename, conf, true)
```
Comment (andrewei1316) [inline mgr/mgr.go]: L797 的赋值导致了回滚的时候，无论之前是 stop 还是  running , 最后都变成了 running，这个地方需要注意一下。另外 reset 后要不要将 stop 状态的 runner 改成 running 状态这个问题，当时我问过崔老师他作为用户怎么想，他说无所谓，所以就维持原来的设定(reset 后自动 running)。但是结合 logkit-pro 的话，我建议此处改为 reset 后维持原状(stop -reset-> stop   running -reset-> running)，这样有利于数据表里面的状态与 agent 的状态保持一致。

Comment (wonderflow) [reply mgr/mgr.go]: logkit-pro这块reset后也变为running，我在pro里改

File: transforms/mutate/trim.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+const (
+	Prefix = "prefix"
+	Suffix = "suffix"
+	Both   = "both"
+)
+
+type Trim struct {
+	Key       string `json:"key"`
+	Character string `json:"characters"`
```
Comment (redHJ) [inline transforms/mutate/trim.go]: Characters

File: transforms/mutate/trim.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
+package mutate
+
+import (
+	"fmt"
+
+	"strings"
```
Comment (redHJ) [inline transforms/mutate/trim.go]: format

**Pull request merged**